Fix journal display of boolean custom fields

NobodysNightmare opened the pull request:
## OpenProject work package

https://community.openproject.org/work_packages/19516
## Description:

Apparently once upon a time there were issues with missing custom value being treated differently from custom values that existed, but were blank.
Thus, at the time some checks using `blank?` and `present?` were introduced, which was fine because custom values are all stringly typed (`"1"` is `true`, `"0"` is `false``).

Fast forward until today, in the APIv3 real dynamic typing is used. Thus setting something to false does set the custom value to `false` rather than `"0"`. ActiveRecord and friends can cope with that, but journals still rely on `blank?` and `present?` which say that `false` is **not present, but blank**.

This leads to missing journal entries for `false` custom values and thus inconsistent journal data.

NobodysNightmare committed:
invert conditions for better readability

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://nobodysnightmare-feature-journal-recognize-false.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://nobodysnightmare-feature-journal-recognize-false.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

NobodysNightmare committed:
style

NobodysNightmare committed:
create CF journal for CF set to false

This is only an issue when the CustomValue.value is actually false.
Legacy code uses stringly typing and thus has values like "1" and "0"

NobodysNightmare committed:
create journal when CF toggles between false + nil

the changed? method of the JournalManager did handle false values
not correctly and considered them blank. However, a boolean CF that is not required
can have (technically) three states:
- true
- false
- nil

Thus we can't consider nil and false as the same case

NobodysNightmare commented:
What's missing here are specs.

I am not really sure how to write good specs, as the `JournalManager` has quite a load of dependencies when it comes to journals of custom fields. An integrative test might be a good idea, do we have a place for that? `spec/models` seems wrong...

myabc commented:
Will ignore Travis failures.
